File: problemas/iniciante/1116/1116.jl
using Printf

n = parse(Int, readline())
for i in 1:n
    numeros = [parse(Float64, x) for x in split(readline())]
    x = numeros[1]
    y = numeros[2]

    if y == 0
        @printf "divisao impossivel\n"
    else
        @printf "%.1f\n" x/y
    end
end

File: problemas/iniciante/1116/1116.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input);
    let n: i32 = input.trim().parse().unwrap();

    for _ in 0..n {
        input = String::new();
        io::stdin().read_line(&mut input);
        let parts: Vec<&str> = input.split(" ").collect();
        let x: i32 = parts[0].trim().parse().unwrap();
        let y: i32 = parts[1].trim().parse().unwrap();

        if y == 0 {
            println!("divisao impossivel");
        } else {
            println!("{:.1}", (x as f64) / (y as f64));
        }
    }
}
